packages/node/src/indexer/store.service.ts

```diff
@@ -14,19 +23,85 @@ export class StoreService {
 
   async syncSchema(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    models: { name: string; attributes: ModelAttributes<any> }[],
+    graphqlSchema: GraphQLSchema,
     schema: string,
   ): Promise<void> {
+    const entitiesRelations = getAllEntitiesRelations(graphqlSchema);
+    const models = getAllEntitiesRelations(graphqlSchema).models.map(
+      (entity) => {
+        const modelAttributes = modelsTypeToModelAttributes(entity);
+        return { name: entity.name, attributes: modelAttributes };
+      },
+    );
     for (const { attributes, name } of models) {
       this.sequelize.define(name, attributes, {
         underscored: true,
+        freezeTableName: false,
         schema,
       });
     }
+
+    const extraQueries = [];
+
+    for (const relation of entitiesRelations.relations) {
+      const model = this.sequelize.model(relation.from);
+      const relatedModel = this.sequelize.model(relation.to);
+
+      if (relation.type === 'belongsTo') {
+        model.belongsTo(relatedModel, { foreignKey: `${relation.foreignKey}` });
+      } else if (relation.type === 'hasOne') {
+        const rel = model.hasOne(relatedModel, {
+          foreignKey: `${relation.foreignKey}`,
+        });
+        const fkConstraint = getFkConstraint(
+          rel.target.tableName,
+          rel.foreignKey,
+        );
+        const tags = smartTags({ singleForeignFieldName: relation.fieldName });
+        extraQueries.push(
+          commentConstraintQuery(
+            `${schema}.${rel.target.tableName}`,
+            fkConstraint,
+            tags,
+          ),
+          createUniqueIndexQuery(
+            schema,
+            relatedModel.tableName,
+            `${relation.foreignKey}`,
+          ),
+        );
+      } else if (relation.type === 'hasMany') {
+        const rel = model.hasMany(relatedModel, {
+          foreignKey: `${relation.foreignKey}`,
+        });
+        const fkConstraint = getFkConstraint(
+          rel.target.tableName,
+          rel.foreignKey,
+        );
+        const tags = smartTags({
+          fieldName: relation.foreignKey.slice(0, -2),
```

Comment:
will we have `id` as foreignKey for one-to-one case? I feel `slice()` is not safe